### ISSUE TITLE
SubscriptionManager not initialized for send-only raw endpoints

### ISSUE THREAD

**Issue by KatoStoelen** [SzymonPobiega/NServiceBus.Raw#10]
https://github.com/SzymonPobiega/NServiceBus.Raw/blob/780d8a1df11c96700e3983f3635a0860df912c77/src/NServiceBus.Raw/InitializableRawEndpoint.cs#L55-L59

The code above is only executed if the endpoint is not send-only. It currently seems like event publishing is not possible for send-only endpoints. But it could have been possible, right? At least for transport with native pub/sub?

What would be the (side) effect of moving the above code outside the `if (!settings.GetOrDefault<bool>("Endpoint.SendOnly"))` block? E.g:

```csharp
if (!settings.GetOrDefault<bool>("Endpoint.SendOnly"))
{
    // Configure receive infrastructure and queue bindings
}

if (transportInfrastructure.OutboundRoutingPolicy.Publishes == OutboundRoutingType.Multicast ||
    transportInfrastructure.OutboundRoutingPolicy.Sends == OutboundRoutingType.Multicast)
{
    subscriptionManager = CreateSubscriptionManager(transportInfrastructure);
}
```

Not sure what "multicast send" means and in which scenarios this is used, but this should work, right? (Without knowing of possible side effects this would yield)

**Comment by SzymonPobiega** [comment]
That should work just fine. `multicast` means that the transport is capable of pub/sub without the need to use an external store to keep the list of subscribers e.g. RabbitMQ or Azure Service Bus.

**Comment by KatoStoelen** [comment]
Yes, I was thinking about this one `transportInfrastructure.OutboundRoutingPolicy.Sends == OutboundRoutingType.Multicast`. I have not seen any scenarions where `Sends` is set to anything other than `Unicast`. Not important, however.

I will test moving the initialization of the subscription manager locally, and submit a PR if it works as expected.

**Comment by SzymonPobiega** [comment]
@KatoStoelen you're right. Currently there is no NSB transport that uses `Multicast` for sends or replies.

**Comment by KatoStoelen** [comment]
After some more digging, I have found out that the problem I'm facing with publishing is not related to the SubscriptionManager not being initialized.

I'm trying to configure a send-only interface in NServiceBus.Router using Azure Service Bus as transport. The SubscriptionManager in the Azure Service Bus transport is only used to create/delete subscriptions, and will throw an exception when trying to initialize it in a send-only scenario. I have mixed some concepts. It's not the SubscriptionManager I need. I just need the send-only interface to forward published events to the topic. Will continue that discussion in SzymonPobiega/NServiceBus.Router#23.

Closing this issue, since the current implementation is as it should be. Thanks for your quick responses.
